### ISSUE TITLE
Standard: light and dark mode

### ISSUE THREAD

**Issue by turkishmountain** [hexchat/hexchat#2444]
I would like that next version of HexChat adds a standard (native integration) light and dark mode in this application. Some users don't know how works it with extra theme in HexChat, than is it difficult for them.

**Comment by CraftSpider** [comment]
Seconding this desire. Even with custom themes, the borders of the panes don't change, so dark themes have incredibly jarring borders

**Comment by ForlornWongraven** [comment]
It's the main reason I close Hexchat in the evening cause I cannot figure out how to do it easily as Gnome can do it just with any other app.

**Comment by xenoterracide** [comment]
+1 and I'm not exactly certain, but I know gnome is, somehow, supposed to be able to communicate mode based on theme; maybe GTK does?. So I would like chat box to be autodetected, getting the default white during a night setup sucks. As long as people can override it. ;)

**Comment by TingPing** [comment]
There is no "dark mode" per-se. However many applications will assume if the theme name ends in `-dark`, like `Adwaita-dark` that you should use a dark mode.

I'm all for this integration but I'm not actively working on it.

Also Windows is a big question mark, it probably has an API for a real "dark-mode" but I don't know.

**Comment by jeois** [comment]
I think we should revisit this issue as Hexchat is one of the most frustrating apps to theme.  In windows, it's not that big of a problem since you're forced to use custom theme plugins.  The installation is a bit of a hassle, but it's doable.

However in linux, any time you change your GTK settings for a theme that doesn't include a separate dark version, it becomes super ugly.  Now that GTK themes are starting to combine both light and dark variations within the same package, Hexchat seems to be one of the few programs that tries to use a combination of both light AND dark when you select color schemes.

For example, the theme "breeze-gtk" now has both light and dark versions in the same package, and you use your desktop environment's settings manager (KDE in my case) to select the color scheme.  This works fine for most GTK programs.  They look great and match everything else.  But in Hexchat, it uses the light colors for the UI, but it does accept that the text background should be dark, so it's an ugly mixture of both. I tried using 'xfce4-settings-manager' and/or 'gnome-tweaks' to change the gtk theme to "Breeze-Dark" but Hexchat is still going by KDE plasma's "Breeze" setting for GTK widget style and only partially applying its color profile, making any theme that does this practically unusable.

**Comment by TingPing** [comment]
> but Hexchat is still going by KDE plasma's "Breeze" setting for GTK widget style and only partially applying its color profile, making any theme that does this practically unusable.

You are confusing completely different things. This happens because HexChat is a GTK2 application and has no concept of a "dark variant" like GTK3 applications.

**Comment by jeois** [comment]
Yes, that is part of the issue.  'gnome-tweaks' offers the dark variant as a separate theme, which doesn't work for hexchat, but KDE presents its color schema selection with at least a dozen variants, without changing the theme:  Some themes are starting to separate color schemes from their widget style.  In any case, both GTK and QT apps look very similar, almost identical, both in color and style.  I suspected GTK3 or 4 included features to make this work easily; but it should be possible, even with GTK2, to pull the selected color scheme and apply it to the UI widgets.  Hexchat is already doing so for the text (i.e. the chat background and user list is properly colored).  Before you think I'm confused again, I know that feature is different from the widgets' theming, but it shows Hexchat is at least accessing the DE's color settings somewhere.

The problem is I have no idea how to do this, which is why I'm asking for help, as I think this dark theme issue doesn't only impact me.  I may be able to fix it for myself, but I'm afraid messing with GTK settings or the color scheme will break it again.  I get that you're volunteers and your time is valuable, and I'd appreciate it if someone eventually took a look at it.  I'll post again if I find a decent workaround.  Thanks for your consideration.

**Comment by TingPing** [comment]
How a dark mode would roughly work:

1. There must be two GTK themes:
    - On Windows this is tricky, but a dark and light theme would have to be bundled.
    - On Linux we can do hacky things like check if `$theme-dark` exists and toggle between them. This would work well for Adwaita and Yaru which are probably the most commonly used themes.
2. There must be two bundled HexChat themes:
    - A "black" theme exists as do other dark ones. I think none of them are default worthy. I think somebody needs to make a new one.
3. HexChat needs to listen to OS specific settings for a dark-mode.
    - On Linux this is a new feature, but it is exposed over dbus via: `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read org.freedesktop.appearance color-scheme`
    - Windows surely has its own global setting for this now.
